utils/nctl/sh/assets/setup.sh

```diff
@@ -86,12 +86,11 @@ function _set_nodes()
             "cfg['rest_server']['address']='0.0.0.0:$(get_node_port_rest "$IDX")';"
             "cfg['rpc_server']['address']='0.0.0.0:$(get_node_port_rpc "$IDX")';"
             "cfg['event_stream_server']['address']='0.0.0.0:$(get_node_port_sse "$IDX")';"
-            "toml.dump(cfg, open('$PATH_TO_FILE', 'w'));"
         )
 
         if [ ! -z "$SPECULATIVE_EXEC_ADDR" ]; then
             SCRIPT+=(
-                "cfg['rpc_server']['speculative_execution_address']='0.0.0.0:$(get_node_port_speculative_exec "$IDX")';"
+                "cfg['speculative_exec_server']['address']='0.0.0.0:$(get_node_port_speculative_exec "$IDX")';"
```

Comment:
Thanks for your response, @Fraser999 :)

I think it could be done this way, but I believe these more 'default' settings values come from the original config.toml, am I mistaken? We could just change the enable_server there and would follow the other servers pattern, what do you think?

[casper-node/resources/local/config.toml](https://github.com/casper-network/casper-node/blob/dev/resources/local/config.toml#L253-L297)

Reply:
That's a good point - probably more overt that way :+1:

Reply:
Thanks for your analysis, can't check the drone output, think I lack authorization to do so, but have seen the tarball.

I agree with your suggestion of always setting the port regardless, I think is the way to go. I was trying to fiddle the less I could with the code, but I share the opinion that this is indeed desirable.

Furthermore, I did this change and also included the setting code in one place that did not follow the same pattern, so I did not find it the first time around. I searched the code for any similar block, and believe I was thorough this time around.

Reply:
So it seems I steered you a little wrong - apologies!  Apparently we do still need to use the `if [ ! -z "$SPECULATIVE_EXEC_ADDR" ]; then` in the `setup_shared.sh` file (updated of course to use `SPECULATIVE_EXEC_ADDR=$(grep 'speculative_exec_server' $PATH_TO_CONFIG_FILE || true)`.

CI effectively runs the following command from the root folder:
```bash
bash -i ./ci/nctl_upgrade.sh
```

I'm running this just now locally using your branch with the change to `setup_shared.sh` I mentioned above, but it takes quite a while to execute, so I'll not be able to see the results until tomorrow.  It has at least got through the first upgrade test now, where previously it was failing.

Reply:
Yes, I think the thing we missed was *upgrade.sh*. After running locally, `bash -i ./ci/nctl_upgrade.sh` I was able to reproduce and hopefully fix this. Putting the same logic into *upgrade.sh* appears to have solved the issue, so I reverted the last commit and did another one changing it. I'll push the changes if you want to give it a try, ok? Cheers

Reply:
Great!  Thanks for sticking with this.  It's passed locally for me too, so I'll try and merge now.

Reply:
Cool! Np, thanks for your assistance! I took the chance to learn a bit more about the whole stack, so it was well worth it. :)